src/main/java/frc/utils/applicationsutils/OutputFile.java

```diff
@@ -0,0 +1,97 @@
+package frc.utils.applicationsutils;
+
+import frc.robot.constants.DirectoryPathsConstants;
+
+import java.awt.*;
+import java.io.File;
+import java.io.FileWriter;
+import java.nio.file.Files;
+import java.nio.file.StandardOpenOption;
+
+public class OutputFile {
+
+    private static final File SELF_OUTPUT_FILE = getSelfOutputFile();
+
+    private final String name;
+    private final File file;
+
+    public OutputFile(String name) {
+        this.name = name;
+        this.file = DirectoryPathsConstants.OUTPUT_FILES_DIRECTORY_PATH.resolve(name).toFile();
+
+        ensureFolderExists();
+        ensureFileExists();
+        clear();
+    }
+
+    private void ensureFolderExists() {
+        File folder = DirectoryPathsConstants.OUTPUT_FILES_DIRECTORY_PATH.toFile();
+        if (!folder.exists()) {
+            folder.mkdir();
+        }
+    }
+
+    private void ensureFileExists() {
+        if (!file.exists()) {
+            createFile();
+        }
+    }
+
+    private void createFile() {
+        try {
+            file.createNewFile();
+        }
+        catch (Exception exception) {
+            reportFileError( "Exception while creating file " + file + "\n" + exception);
+        }
+    }
+
+    public void write(String text) {
+        try {
+            Files.writeString(file.toPath(), text + "\n", StandardOpenOption.APPEND);
+        }
+        catch (Exception exception) {
+            reportFileError("Unable to write to output file: " + name + "\n" + exception);
+        }
+    }
+
+    public void open() {
+        try {
+            if (!Desktop.isDesktopSupported()) {
+                reportFileError("Desktop is not supported on this platform");
+            }
+            else if (file.exists()) {
+                Desktop.getDesktop().open(file);
+            }
+        }
+        catch (Exception exception) {
+            reportFileError("Exception While Opening File " + file + "\n" + exception);
+        }
+    }
```

Comment:
Replace to Files functionality

Reply:
there is no `Files.open()`

Reply:
Yes I learn only the exist, it’s fine now